pennylane/devices/task_qubit.py

```diff
@@ -0,0 +1,243 @@
+# Copyright 2018-2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""This module contains a proxy qubit object for spawning multiple instances of 
+a given qubit type for run on a background scheduler.
+"""
+from typing import List, Union, Tuple, Dict
+from contextlib import nullcontext
+import pennylane as qml
+from pennylane import QubitDevice
+from .default_qubit import DefaultQubit
+from .._version import __version__
+
+try:
+    import dask
+    import dask.distributed as dist
+    from dask.distributed import worker_client, performance_report
+
+    from distributed.protocol import (
+        dask_serialize,
+        dask_deserialize,
+    )
+
+    @dask_serialize.register(qml.numpy.tensor)
+    def serialize(tensor: qml.numpy.tensor) -> Tuple[Dict, List[bytes]]:
+        "Defines a serializer for the Dask backend"
```

Comment:
This is quite nice!

Although, I worry that this might lull users into a false sense of security; that is, that autograd works (however, they will later find it only works on the forward pass!)

Does it make sense to simply not support autograd, and require users to use vanilla NumPy for the task device?

Or is this required due to assumptions made in other parts of PL, where autograd arrays end up in the device even without the user explicitly doing it?

Reply:
Good question: I think these can be removed as we discovered the autograd elements don't work for all cases. Numpy, TF Tensors and PyTorch too all have supported types in Dask, so we shouldn't need to handle any ser/deser functions for them. Though, I do see this being necessary later once the https://github.com/xanaduAI/xir becomes more feature-complete, as we can pass the instructions rather than the object data to workers. I'll remove these for now, and revisit when XIR integration can commence.

Reply:
If it will be useful later, feel free to leave it in!

Reply:
We just need to be clear in the documentation that gradients are not supported with Autograd. Additionally, perhaps we could intercept the known error message with a better one just so users who don't read the documentation aren't confused.

Reply:
I have added comments on avoiding the use of `autograd` here. I don't see a clean way of keeping the logic within the `TaskQubit` class to raise a warning/error, as I think this requires modifications to the `Qnode` to track appropriately. Though, this will likely become an if-else nightmare if we intend to provide many differently supported conditions. I need to think of a better solution to handle this cleanly.